pkg/protocol/recordlayer/recordlayer.go

```diff
@@ -85,11 +82,40 @@ func UnpackDatagram(buf []byte) ([][]byte, error) {
 	out := [][]byte{}
 
 	for offset := 0; len(buf) != offset; {
-		if len(buf)-offset <= HeaderSize {
+		if len(buf)-offset <= DefaultHeaderSize {
+			return nil, errInvalidPacketLength
+		}
+
+		pktLen := (DefaultHeaderSize + int(binary.BigEndian.Uint16(buf[offset+11:])))
+		if offset+pktLen > len(buf) {
+			return nil, errInvalidPacketLength
+		}
+
+		out = append(out, buf[offset:offset+pktLen])
+		offset += pktLen
+	}
+
+	return out, nil
+}
+
+// ContentAwareUnpackDatagram is the same as UnpackDatagram but considers the
+// presence of a connection identifier if the record is of content type
+// tls12_cid.
+func ContentAwareUnpackDatagram(buf []byte, cidLength int) ([][]byte, error) {
+	out := [][]byte{}
+
+	for offset := 0; len(buf) != offset; {
+		headerSize := DefaultHeaderSize
+		lenIdx := 11
```

Comment:
Worth making `11` a constant and dropping a visualization of the header with it?